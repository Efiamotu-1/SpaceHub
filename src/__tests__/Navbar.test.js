import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Navbar component', () => {
  test('Correct render of Navbar', () => {
    const navbar = render(
      <Router>
        <NavBar />
      </Router>,
    );
    expect(navbar).toMatchSnapshot();
  });

  test('Check if \'Space Traveler\'s Hub\' is in the rendered document of Navbar', () => {
    render(
      <Router>
        <NavBar />
      </Router>,
    );
    expect(screen.getByText('Space Traveler\'s Hub')).toBeInTheDocument();
  });
});
