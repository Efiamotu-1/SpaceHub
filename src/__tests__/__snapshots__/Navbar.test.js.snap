// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Correct render of Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="md:flex justify-between mx-5 px-5 py-2 bg-gray-00 items-center border-b border-b-gray-200"
      >
        <div
          class="flex items-center"
        >
          <img
            alt="logo"
            class="w-20 mr-2"
            src="space-img.png"
          />
          <p
            class="text-3xl"
          >
            Space Traveler's Hub
          </p>
        </div>
        <div>
          <div
            class="flex space-x-14 items-center mx-5"
          >
            <ul
              class="flex space-x-14 text-2l text-blue-500"
            >
              <li>
                <a
                  aria-current="page"
                  class="hover:text-stone-900 active:text-gray-400 active"
                  href="/"
                >
                  ROCKETS
                </a>
              </li>
              <li>
                <a
                  class="hover:text-stone-900 active:text-gray-400"
                  href="/missions"
                >
                  MISSIONS
                </a>
              </li>
              <li>
                <a
                  class="hover:text-stone-900 active:text-gray-400"
                  href="/my-profile"
                >
                  MY PROFILE
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="md:flex justify-between mx-5 px-5 py-2 bg-gray-00 items-center border-b border-b-gray-200"
    >
      <div
        class="flex items-center"
      >
        <img
          alt="logo"
          class="w-20 mr-2"
          src="space-img.png"
        />
        <p
          class="text-3xl"
        >
          Space Traveler's Hub
        </p>
      </div>
      <div>
        <div
          class="flex space-x-14 items-center mx-5"
        >
          <ul
            class="flex space-x-14 text-2l text-blue-500"
          >
            <li>
              <a
                aria-current="page"
                class="hover:text-stone-900 active:text-gray-400 active"
                href="/"
              >
                ROCKETS
              </a>
            </li>
            <li>
              <a
                class="hover:text-stone-900 active:text-gray-400"
                href="/missions"
              >
                MISSIONS
              </a>
            </li>
            <li>
              <a
                class="hover:text-stone-900 active:text-gray-400"
                href="/my-profile"
              >
                MY PROFILE
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
